AxonDeepSeg/data_management/input_data.py

```diff
@@ -268,36 +275,37 @@ def next_batch_WithWeights(self, augmented_data_ = {'type':'None'},
 
             image, real_mask, real_weights = self.apply_data_augmentation([image, real_mask, real_weights],
                                                                           augmented_data, data_aug_verbose)
-            
+
             # Normalisation of the image
             image = apply_legacy_preprocess(image)
 
             # We have now loaded the good image, a mask (under the shape of a matrix, with different labels) that still needs to be converted to a volume (meaning, a sparse cube where each layer of depth relates to a class)
-            
+
             batch_x.append(image)
             batch_y.append(real_mask)
             batch_w.append(real_weights)
-            
+
             # If we are at the end of an epoch, we reset the list of samples, so that during next epoch all sets will be different.
             if self.sample_seen == self.epoch_size:
                 if each_sample_once:
                     self.samples_list = self.reset_set(type_ = 'validation')
                     break
                 else:
                     self.samples_list = self.reset_set(type_ = 'train')
-                    
 
-        # Ensuring that we do have np.arrays of the good size for batch_x and batch_y before returning them        
+
+        # Ensuring that we do have np.arrays of the good size for batch_x and batch_y before returning them
         return transform_batches([batch_x, batch_y, batch_w])
-    
+
     def read_image(self, type_, i):
         '''
         :param i: indice of the image or mask to read.
-        :return image: the loaded image with 8 bit pixels, range of values being [0,288] 
+        :return image: the loaded image with 8 bit pixels, range of values being [0,288]
         '''
 
         # Loading the image using 8-bit pixels (0-255)
-        return imread(os.path.join(self.path,str(type_) + '_%s.png' % i), flatten=False, mode='L')
+        filename = str(type_) + '_%s.png' % i
```

Comment:
Yep, you got around `joinpath` here similar to how I recommended in the file above.